llvm/lib/Target/BPF/BPFAbstractMemberAccess.cpp

```diff
@@ -1113,8 +1113,9 @@ bool BPFAbstractMemberAccess::transformGEPChain(CallInst *Call,
                               Call->getIterator());
 
   // Generate a BitCast
-  auto *BCInst =
-      new BitCastInst(Base, PointerType::getUnqual(BB->getContext()));
+  auto *BCInst = new BitCastInst(
+      Base, PointerType::get(BB->getContext(),
+                             Base->getType()->getPointerAddressSpace()));
```

Comment:
Can't you drop this bitcast and the one below entirely? They are not needed anymore with opaque pointers.

Reply:
Okay, let me give a try.

Reply:
Did this change make it into the backport: https://github.com/llvm/llvm-project/pull/130995

Reply:
Since backport is only for bug fix, I think the suggestion to drop the above bitcast does not need to be backported since it does not fix any bug. I hope my understanding is correct. But for llvm21, I will make proper changes to remove these bitcasts.